src/librawspeed/metadata/ColorFilterArray.h

```diff
@@ -27,16 +27,15 @@
 namespace RawSpeed {
 
 enum CFAColor {
-  CFA_COLOR_MIN = 0,
+  // see also DngDecoder
   CFA_RED = 0,
   CFA_GREEN = 1,
   CFA_BLUE = 2,
-  CFA_GREEN2 = 3,
+  CFA_INVALID = 3, // keep the unit tests happy
```

Comment:
> That is the code that is used by darktable, dt_rawspeed_crop_dcraw_filters()
> So it is needed, i think.

Right. I could have named it "dummy" or "DCRAW_FILTER_HELPER_THINGY" ;). I think the issues come from the fact that it is possible to go from the enum to dcraw_filter in a defined way, but not the other way around (the mapping is not bijective). So the 'correct' approach would be to remove the constructor taking a dcraw_filter and instead provide a little helper function that does perform the bit-swiveling directly on the uint32. And then probably put that code directly in the dt-code you linked above. That way, we can completely do away with both the INVALID as well as the FUJI_GREEN color and use integers that directly map to the ones used in the CFAPATTERN fields of the raw files. See also the DngDecoder int->enum mapping.